### PR TITLE
refactor: align arguments in json service

### PR DESCRIPTION
**Pull request by pamapa**
As suggested in MR #338. To be merged after #338...

### Checklist

- [ ] This PR makes changes to the public API 
- [ ] I have included links for closing relevant issue numbers

### CODE REVIEW

File: src/JsonService.ts [hunk 1]
```diff
@@ -74,7 +122,12 @@ export class JsonService {
         return json;
     }
 
-    public async postForm(url: string, body: URLSearchParams, basicAuth?: string): Promise<Record<string, unknown>> {
+    public async postForm({
+        url,
```
Comment (kherock) [inline src/JsonService.ts:126]: I don't have a particularly strong opinion on the API here, but I think keeping `url: string` for the first arguments would be nice it that it mirrors `fetch` and would make many of the tests simpler to write.

File: src/TokenClient.test.ts [hunk 2]
```diff
@@ -87,9 +87,9 @@ describe("TokenClient", () => {
             expect(generateBasicAuthSpy).toHaveBeenCalledWith("client_id", "client_secret");
             expect(getTokenEndpointMock).toBeCalledWith(false);
             expect(postFormMock).toBeCalledWith(
-                "http://sts/token_endpoint",
-                expect.any(URLSearchParams),
-                expect.stringContaining(""),
+                expect.objectContaining({
+                    url: "http://sts/token_endpoint",
+                }),
```
Comment (kherock) [inline src/TokenClient.test.ts:92]: This test is a less strict than it was before, the previous revision had checks for URLSearchParams body and the basic auth string. There's several other places within this file of this occurring.

**Review by kherock — APPROVED** [review]
Nice, thank you

**Comment by pamapa** [comment]
Thanks for reviewing, it is much better like this :-)

**Pull request merged**